### PR TITLE
Flower page dev

### PR DESCRIPTION
**Pull request by aleksandranedic**

### CODE REVIEW

File: front/my-app/src/FlowerCards/FlowerInfo.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+interface FlowerInfoProps {
+    data: FlowerDetails;
+}
+ 
+const FlowerInfo: React.FunctionComponent<FlowerInfoProps> = ({data}) => {
+    return ( 
+        <>
+        <div className="flex flex-col sm:flex-row p-10 gap-10 sm:gap-64 lg:gap-80">
+                <div className="text-secondary-gray font-Ubuntu !font-bold text-sm flex gap-y-2 flex-col items-start sm:mt-36">
+                    {data.flower.features.length > 0 && (<><p>Features:</p>{data.flower.features.map((feature) => (<p key={feature}>{feature}</p>))}</>)}
```
Comment (kiljee) [inline front/my-app/src/FlowerCards/FlowerInfo.tsx:12]: object for key?

File: front/my-app/src/utils/FavoriteButton.tsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { AiFillStar } from "react-icons/ai";
+import { Flower, setFavorite } from "../model/FlowerInterface";
+import { useAppSelector } from "../state/hooks";
+
+interface FavoriteButtonProps {
+    flower: Flower;
+}
+ 
+const FavoriteButton: React.FunctionComponent<FavoriteButtonProps> = ({flower}) => {
+    const loged = useAppSelector((state) => state.logedUser.loged);
```
Comment (kiljee) [inline front/my-app/src/utils/FavoriteButton.tsx:10]: naming logged or loged
({logedUser: { loged}}=> {loged})

File: front/my-app/src/model/UserInfo.tsx [hunk 3]
```diff
@@ -57,11 +48,7 @@ export class User {
     return this.date_of_birth;
   }
 
-  get sightingsNum() {
-    return this.sightings.length;
-  }
-
-  private getSightings = async (id: number) => {
+  /*private getSightings = async (id: number) => {
```
Comment (kiljee) [inline front/my-app/src/model/UserInfo.tsx:51]: delete comments

File: front/my-app/src/modals/LoginSuccessModal.tsx [hunk 4]
```diff
@@ -25,18 +27,28 @@ const LoginSuccessModal: React.FunctionComponent<ModalProps> = ({
...
 
+  const fetchUserSightingsDetails = (arr:Sighting[]) => {
+    let sightings:SightingDetails[] = []
+    arr.forEach((element:Sighting) => pushData(sightings, element));
+    return sightings;
+  }
+
+  const pushData = async(sightings:SightingDetails[],element:Sighting) => {
+    let data:SightingDetails = await fetchSightingInfo<SightingInfoData>(`sightings/${element.id}`).then(r => r.data.sighting);
+    console.log("***")
```
Comment (kiljee) [inline front/my-app/src/modals/LoginSuccessModal.tsx]: remove console.log

File: front/my-app/src/FlowerCards/Favorites.tsx [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import FlowerCardsDisplay from "./FlowerCardsDisplay";
+import { useAppSelector } from "../state/hooks";
+import { FavoriteFlower } from "../model/FlowerInterface";
+
+const Favorites: React.FunctionComponent = () => {
+  const flowers:FavoriteFlower[] = useAppSelector((state) => state.logedUser.favoriteFlowers.flowers);
```
Comment (kiljee) [inline front/my-app/src/FlowerCards/Favorites.tsx:7]: Can use destructors here

File: front/my-app/src/FlowerCards/FlowerBanner.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+const FlowerBanner: React.FunctionComponent<FlowerBannerProps> = ({data}) => {
+    return ( <>
+        <div className="relative w-full h-96 sm:h-fit">
+                <img src="../images/flower-banner.png" alt="banner" className="banner h-96 object-contain hidden sm:flex w-full"/>
+      
+                <img src={data.flower.profile_picture} alt="profile" className="absolute w-full h-96 object-fill sm:top-32 sm:left-4 sm:rounded sm:w-80 brightness-50 sm:filter-none"/>
+
+                <div className="flex gap-x-5 gap-y-3 justify-start flex-wrap w-fit absolute top-1/2 left-6 sm:top-40 sm:left-1/2 lg:left-1/4 lg:ml-10 lg:top-1/2">
+                    <FavoriteButton flower={data.flower} />
+                    <SightingsButton favorite={data.flower.favorite} sightings={data.flower.sightings} />
```
Comment (kiljee) [inline front/my-app/src/FlowerCards/FlowerBanner.tsx:19]: formating

File: front/my-app/src/FlowerCards/FlowerCard.tsx [hunk 7]
```diff
@@ -1,38 +1,32 @@
...
+import { FlowerData } from "../model/FlowerInterface";
+import FavoriteButton from "../utils/FavoriteButton";
+import SightingsButton from "../utils/SightingsButton";
 
-const FlowerCard: React.FC<Flower> = ({ flower }) => {
-  const loged = useAppSelector((state) => state.logedUser.loged);
+
+const FlowerCard: React.FC<FlowerData> = ({ flower }) => {
+
+  function seeProfile(id:number) {
```
Comment (kiljee) [inline front/my-app/src/FlowerCards/FlowerCard.tsx:9]: why using function use arrow function

File: front/my-app/src/Sightings/SightingDetails.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+import SightingDetailsInfo from "./SightingDetailsInfo";
+
+ 
+const SightingDetailsComp: React.FunctionComponent = () => {
+    const { id } = useParams<{id:string}>();
+    const [data, isPending, error] = useFetch<SightingInfoData>(fetchSightingInfo,`sightings/${id}`)
+    const [sighting, setSighting] = useState<SightingDetails | null>(null);
+
+    useEffect(() => {
+        if (typeof data !== "undefined") {
```
Comment (kiljee) [inline front/my-app/src/Sightings/SightingDetails.tsx:17]: if(!data) return;

File: front/my-app/src/Sightings/SightingDetailsComments.tsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    const [comment, setComment] = useState<string>("");
+    const loged:boolean = useAppSelector(state => state.logedUser.loged)
+    const [data, isPending, error] = useFetch<sightingCommentsData>(fetchSightingComments, `sightings/${id}/comments?page=${currPage}`)
+    const [comments, setComments] = useState<SightingComment[]>([]);
+
+    const commentRef = useRef<HTMLTextAreaElement>(null);
+
+    const  handleSubmit = async (event:FormEvent) => {
+        event.preventDefault();
+        if (comment) {
```
Comment (kiljee) [inline front/my-app/src/Sightings/SightingDetailsComments.tsx:26]: if(!comment) return;

File: front/my-app/src/Sightings/SightingsBanner.tsx [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+const SightingsBanner: React.FunctionComponent = () => {
+    return (    
+        <div className="flex justify-center w-full p-10 relative">
+            <div className="flex flex-col items-center font-Ubuntu !font-thin text-secondary-gray ">
+                <p className="text-4xl"> Sightings list </p>
+                <p className="text-lg"> Explore between more that 8427 sightings </p>
+            </div>
+           
+            <div className="hidden sm:flex absolute sm:right-10 sm:bottom-16">
+                {/* <AddSightingButton /> */}
```
Comment (kiljee) [inline front/my-app/src/Sightings/SightingsBanner.tsx:13]: remove comment

File: front/my-app/src/Sightings/SightingUserInfo.tsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+                        <div className="flex flex-col justify-start">
+                            <p className="font-Ubuntu text-secondary-title">{flowerName}</p>
+                            <p className="font-Ubuntu text-secondary-gray text-sm italic"> By {userFullName}</p>
+                        </div>
+                    </div>
+                    <div className={`cursor-default font-Montserrat !font-medium text-secondary-gray text-justify tracking-wider ${window.location.href.includes("sighting/") ? "break-words" : "truncate"}`}>
+                        {description}
+                    </div>
+                    <hr/>
+                    {(typeof commentsCount !== "undefined") && (typeof likesCount !== "undefined") && <div className="flex justify-start gap-7 font-Ubuntu text-secondary-gray">
```
Comment (kiljee) [inline front/my-app/src/Sightings/SightingUserInfo.tsx:25]: {( commentsCount && likesCount &&

File: front/my-app/src/fetch-data/services/sightingService.ts [hunk 12]
```diff
@@ -1,14 +1,40 @@
...
+  return await http.post(url, data)
+}
+export const fetchSightingComments = async (url: string): Promise<AxiosResponse> => {
+  return await http.get<sightingCommentsData, Promise<AxiosResponse>>(url);
+}; 
+
+export const addComment = async (url: string, data:addCommentData): Promise<AxiosResponse> => {
+  return await http.post(url, data)
+}
 /*export const createUser = async (user: User): Promise<User> => {
```
Comment (kiljee) [inline front/my-app/src/fetch-data/services/sightingService.ts:38]: remove comments

File: front/my-app/src/FlowerCards/FlowerInfo.tsx (hunk 1 above)
Comment (aleksandranedic) [reply front/my-app/src/FlowerCards/FlowerInfo.tsx:12]: Key is variable of type string

**Pull request closed without merging**